wtxmgr/tx.go

```diff
@@ -898,8 +988,46 @@ func getP2PKHOpCode(pkScript []byte) uint8 {
 	return OP_NONSTAKE
 }
 
+// pkScriptType determines the general type of pkScript for the purposes of
+// fast extraction of pkScript data from a raw transaction record.
+func pkScriptType(pkScript []byte) (scriptType, error) {
+	var scrType scriptType
```

Comment:
I would just return inside the switch statement, so that the switch statement becomes a terminating statement (needs a default case too).  If a new script class is added and not all cases are handled, this will return the zero value with no error, which is probably wrong.